sound/soc/sof/ipc4-pcm.c

```diff
@@ -916,46 +937,108 @@ static snd_pcm_sframes_t sof_ipc4_pcm_delay(struct snd_soc_component *component,
 	if (time_info->stream_start_offset == SOF_IPC4_INVALID_STREAM_POSITION) {
 		ret = sof_ipc4_get_stream_start_offset(sdev, substream, stream, time_info);
 		if (ret < 0)
-			return 0;
+			return -EOPNOTSUPP;
```

Comment:
should all the changes in error handling be a different patch?

Reply:
Unfortunately not, the definition of the pointer callback is that we will ignore fallback to the default pointer code in case -EOPNOTSUPP is returned just like if the callback itself is not set.
If we cannot get things for the delay we shall still report the pointer. We might have real errors to report or other archs will have.